cpp/include/cudf/join/hash_join.hpp

```diff
@@ -242,6 +243,82 @@ class hash_join {
     rmm::cuda_stream_view stream      = cudf::get_default_stream(),
     rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref()) const;
 
+  /**
+   * @brief Returns context information about matches between the probe and build tables.
+   *
+   * This method computes, for each row in the probe table, how many matching rows exist in
+   * the build table according to inner join semantics, and returns the number of matches through a
+   * join_match_context object.
+   *
+   * This is particularly useful for:
+   * - Determining the total size of a potential join result without materializing it
+   * - Planning partitioned join operations for large datasets
+   *
+   * @param probe The probe table to join with the pre-processed build table
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   * @param mr Device memory resource used to allocate the result device memory
+   *
+   * @throw cudf::logic_error If the input probe table has nulls while this hash_join object was not
+   * constructed with null check.
```

Comment:
Can we move the `@throw` above the `@param` section?
https://github.com/rapidsai/cudf/blob/branch-25.10/cpp/doxygen/developer_guide/DOCUMENTATION.md#the-example